### ISSUE TITLE
fix error with null document.body

### ISSUE THREAD

**Issue by AndreyMiloserdov** [react-bootstrap/dom-helpers#29]
In some libraries used dom-helpers appear error:

```
matches.js:23 Uncaught TypeError: Cannot read property 'matches' of null
    at matches.js:23
    at Object.exports.__esModule (matches.js:28)
    at __webpack_require__
```

just replace code in dom-helpers/query/matches.js from

```
  if (_inDOM2.default) {
    (function () {
      var body = document.body;
      var nativeMatch = body.matches || body.matchesSelector || body.webkitMatchesSelector || body.mozMatchesSelector || body.msMatchesSelector;

      matches = nativeMatch ? function (node, selector) {
        return nativeMatch.call(node, selector);
      } : ie8MatchesSelector;
    })();
  }
```

to

```
document.addEventListener('DOMContentLoaded', function() {
  if (_inDOM2.default) {
    (function () {
      var body = document.body;
      var nativeMatch = body.matches || body.matchesSelector || body.webkitMatchesSelector || body.mozMatchesSelector || body.msMatchesSelector;

      matches = nativeMatch ? function (node, selector) {
        return nativeMatch.call(node, selector);
      } : ie8MatchesSelector;
    })();
  }
});
```

That fix problems for null document.body

**Comment by vdh** [comment]
I tried applying that fix but it failed the Karma tests.

This whole matches functionality is written synchronously around the idea that `DOMContentLoaded` has already fired. I can't think of a good way to fix it without rewriting the whole thing to properly wait asynchronously for `DOMContentLoaded`.

**Comment by jquense** [comment]
Yes you should either load scripts in the body per web best practices or do the content loaded handler at the root of your app

**Comment by vdh** [comment]
@jquense I know about best practices, but they shouldn't then get turned into silent hard dependencies.

I already have a `DOMContentLoaded` listener before the root of my app, but because this async dependency is happening at `require` time, this `matches` code is executing _immediately_, long before any React component code that actually needs to use it is fired… :(

**Comment by jquense** [comment]
Yes, there is basically a dependency that scripts 
it execute before the body. You don't need to wait for content loaded, moving the script tag to below the body will work as well.  I'm happy to take a PR switching all checks to something more lazy and keeps the API the same. It may be easier tho to not load the script in the HEAD
